Challenge 1 - Inventory Dates - Completed

rutwick33 opened the pull request:
Changes made to the inventory.views.py and added the separate url

rutwick-tmt committed:
Challenge 1 - Inventory Dates

Changes made to the Inventory views.py and added the separate url

b123m commented on interview/inventory/urls.py:
```diff
@@ -1,6 +1,10 @@
 
 from django.urls import path
-from interview.inventory.views import InventoryLanguageListCreateView, InventoryLanguageRetrieveUpdateDestroyView, InventoryListCreateView, InventoryRetrieveUpdateDestroyView, InventoryTagListCreateView, InventoryTagRetrieveUpdateDestroyView, InventoryTypeListCreateView, InventoryTypeRetrieveUpdateDestroyView
+from interview.inventory.views import InventoryLanguageListCreateView, \
```
slashes here are considered bad practice. If you are going to separate them onto different lines like this try this format:
from x import (
y,
z,
w
)

rutwick33 replied:
Changes made.

b123m commented on interview/inventory/urls.py:
```diff
@@ -12,5 +16,6 @@
     path('languages/', InventoryLanguageListCreateView.as_view(), name='inventory-languages-list'),
     path('tags/', InventoryTagListCreateView.as_view(), name='inventory-tags-list'),
     path('types/', InventoryTypeListCreateView.as_view(), name='inventory-types-list'),
+    path('day/', InventoryPostDayListCreateView.as_view(), name='inventory-day-list'),
```
Is this a good naming convention for the url?

rutwick33 replied:
Changes made.

b123m commented on interview/inventory/views.py:
```diff
@@ -63,6 +63,29 @@ def delete(self, request: Request, *args, **kwargs) -> Response:
         return self.queryset.get(**kwargs)
+    
+    
+class InventoryPostDayListCreateView(APIView):
```
This name is confusing

rutwick33 replied:
Changes made.

b123m commented on interview/inventory/views.py:
```diff
@@ -63,6 +63,29 @@ def delete(self, request: Request, *args, **kwargs) -> Response:
+        try:
+            
+            # Inventory items created after this day will be shown 
+            start_date_1 = dt.datetime(2023,3,3)
```
Why is this hard coded? The instructions asked for this to be production code. Would this function in production?

rutwick33 replied:
Changes made.

b123m commented on interview/inventory/views.py:
```diff
@@ -63,6 +63,29 @@ def delete(self, request: Request, *args, **kwargs) -> Response:
+            start_date_1 = dt.datetime(2023,3,3)
+            
+            inventory = self.get_queryset(created_at__gte=start_date_1)
+            print(inventory.values())
```
Print statements are bad practice in production django code. This should be a logger.

rutwick33 replied:
Changes made.

b123m commented on interview/inventory/views.py:
```diff
@@ -63,6 +63,29 @@ def delete(self, request: Request, *args, **kwargs) -> Response:
+            return Response(serializer.data, status=200)
+        
+        except Exception as e:
+            print("******** Exception: ",e)
```
print statement. see above

rutwick33 replied:
Changes made.

b123m reviewed (commented):
Great first attempt. Please make the corrections from the comments. You also need to add testing so we can validate whether this works.

b123m reviewed (changes requested):
Great first attempt. Please make the corrections from the comments. You also need to add testing so we can validate whether this works.

rutwick-tmt committed:
Following changes made

Updated the import statement and removed the slashes
Changed the Name of the URL path  Changed the name of the view
Removed the hardcoded date
Added Logger
Added a simple test
